bluesky/utils.py

```diff
@@ -14,12 +16,16 @@
 import abc
 from collections.abc import Iterable
 import numpy as np
+from IPython.core.display import display, HTML
```

Comment:
This module needs to be importable without IPython installed. (IPython is an optional requirement.) We could move this to a separate module or we could just hide these imports inside the functions methods where they are used. I'd be inclined toward the latter in this case, but either is fine by me.

Ditto for the other new imports below.

Reply:
I usually favour many small modules so I'd prefer move it if that's okay. 
Do you think there is scope for creating a notebook utility module and moving NotebookProgressBar into it?

Reply:
Yes. This `utils.py` module has grown accidentally huge. Might be a good moment to create a `bluesky.utils` sub-package and move this file into it in a backward-compatible fashion so we can carve it up at our leisure, as in:

```py
# bluesky/utils/__init__.py
from ._legacy_utils import *
```

Then perhaps this could go into `bluesky.utils.jupyter`.

Reply:
I've had a go, let me know what you think.
A wildcard import like you suggest did not work because some modules import private functions from utils.
Instead I put the legacy code into utils/__init__.py